compiler-utils/src/testFixtures/java/com/squareup/anvil/compiler/internal/testing/AnvilCompilation.kt

```diff
@@ -48,41 +55,68 @@ public class AnvilCompilation internal constructor(
       // Deprecation tracked in https://github.com/square/anvil/issues/672
       @Suppress("DEPRECATION")
-      componentRegistrars = listOf(
-        AnvilComponentRegistrar().also { it.addCodeGenerators(codeGenerators) }
-      )
+      componentRegistrars = listOf(anvilComponentRegistrar)
       if (enableDaggerAnnotationProcessor) {
         annotationProcessors = listOf(ComponentProcessor(), AutoAnnotationProcessor())
       }
 
       val anvilCommandLineProcessor = AnvilCommandLineProcessor()
       commandLineProcessors = listOf(anvilCommandLineProcessor)
 
-      pluginOptions = listOf(
-        PluginOption(
-          pluginId = anvilCommandLineProcessor.pluginId,
-          optionName = "src-gen-dir",
-          optionValue = File(workingDir, "build/anvil").absolutePath
-        ),
+      pluginOptions = mutableListOf(
         PluginOption(
           pluginId = anvilCommandLineProcessor.pluginId,
-          optionName = "generate-dagger-factories",
-          optionValue = generateDaggerFactories.toString()
-        ),
-        PluginOption(
-          pluginId = anvilCommandLineProcessor.pluginId,
-          optionName = "generate-dagger-factories-only",
-          optionValue = generateDaggerFactoriesOnly.toString()
+          optionName = "disable-component-merging",
+          optionValue = disableComponentMerging.toString()
         ),
         PluginOption(
           pluginId = anvilCommandLineProcessor.pluginId,
-          optionName = "disable-component-merging",
-          optionValue = disableComponentMerging.toString()
+          optionName = "compilation-mode",
+          optionValue = mode.type.name.lowercase(Locale.US)
         )
       )
 
+      when (mode) {
+        is Embedded -> {
+          anvilComponentRegistrar.addCodeGenerators(mode.codeGenerators)
+          pluginOptions +=
+            listOf(
+              PluginOption(
+                pluginId = anvilCommandLineProcessor.pluginId,
+                optionName = "src-gen-dir",
+                optionValue = File(workingDir, "build/anvil").absolutePath
+              ),
+              PluginOption(
+                pluginId = anvilCommandLineProcessor.pluginId,
+                optionName = "generate-dagger-factories",
+                optionValue = generateDaggerFactories.toString()
+              ),
+              PluginOption(
+                pluginId = anvilCommandLineProcessor.pluginId,
+                optionName = "generate-dagger-factories-only",
+                optionValue = generateDaggerFactoriesOnly.toString()
+              )
+            )
+        }
+        is Ksp -> {
+          symbolProcessorProviders += buildList {
+            addAll(
+              ServiceLoader.load(
+                SymbolProcessorProvider::class.java,
+                SymbolProcessorProvider::class.java.classLoader
+              )
+            )
+            addAll(mode.symbolProcessorProviders)
+          }
+          kspWithCompilation = true
```

Comment:
This is how we force KSP to compile in-process during the kotlinc invocation, just for testing.

Reply:
It's on the kdoc iirc